### DISCUSSION TITLE
Redirect to /subpage after receiving post request

### DISCUSSION THREAD

**Discussion by AndiLeni** [fastapi/fastapi#2789] [Questions] [answered]
Hello,

I have some troubles with redirecting the user.
I use fastapi's jinja module, so the user gets a full website to see.

This is my setup:
- `/add` accepts GET to display a page and POST to add a new item
- `/list` accepts GET to show all my items

After adding a new item on /add, I want to redirect the user to the /list page, unfortunately this gives me a `405 Method Not Allowed` response.

It seems like fastapi tries to create a POST request to /list.
The log says:
````
INFO:     127.0.0.1:53116 - "POST /add HTTP/1.1" 307 Temporary Redirect
INFO:     127.0.0.1:53116 - "POST /list HTTP/1.1" 405 Method Not Allowed``
````

The redirect happens with:
````
@app.post("/add")
async def create_file(file: UploadFile = File(...)):
   ...
    return RedirectResponse("/list")
````

How can I tell the RedirectRespons to create a GET request?
Or any other way so solve this?

Thanks in advance!
Andreas

**Comment by frankie567** [comment, marked as answer]
This is not FastAPI that triggers a POST request, it's the browser that automatically follows the redirection with the same method.

By default, `RedirectResponse` answers with a 307 status code. In your case, you want to implement the [Post/Redirect/Get pattern](https://en.wikipedia.org/wiki/Post/Redirect/Get) which expects a 303 status code ; so that the browser knows to do a GET request after.

So this will work:

```py
return RedirectResponse("/list", status_code=303)
```

**Reply by AndiLeni** [reply]
Thanks, this works for me!
